osu.Server.Spectator.Tests/Multiplayer/MultiplayerAllPlayersRoundRobinQueueTests.cs

```diff
@@ -13,125 +14,81 @@ namespace osu.Server.Spectator.Tests.Multiplayer
+        public async Task RoundRobinOrdering()
         {
             Database.Setup(d => d.GetBeatmapChecksumAsync(3333)).ReturnsAsync("3333");
-            Database.Setup(d => d.GetBeatmapChecksumAsync(4444)).ReturnsAsync("4444");
-            Database.Setup(d => d.GetBeatmapChecksumAsync(5555)).ReturnsAsync("5555");
 
             await Hub.JoinRoom(ROOM_ID);
             await Hub.ChangeSettings(new MultiplayerRoomSettings { QueueMode = QueueMode.AllPlayersRoundRobin });
 
-            await Hub.AddPlaylistItem(new MultiplayerPlaylistItem
-            {
-                BeatmapID = 3333,
-                BeatmapChecksum = "3333"
-            });
-
-            await Hub.AddPlaylistItem(new MultiplayerPlaylistItem
-            {
-                BeatmapID = 4444,
-                BeatmapChecksum = "4444"
-            });
-
             SetUserContext(ContextUser2);
-
             await Hub.JoinRoom(ROOM_ID);
 
-            await Hub.AddPlaylistItem(new MultiplayerPlaylistItem
-            {
-                BeatmapID = 5555,
-                BeatmapChecksum = "5555"
-            });
-
+            SetUserContext(ContextUser);
             checkCurrentItem(1);
 
-            await runGameplay();
-            checkCurrentItem(4);
-
-            await runGameplay();
-            checkCurrentItem(2);
-
-            await runGameplay();
-            checkCurrentItem(3);
-        }
-
-        [Fact]
-        public async Task CurrentItemUpdatedWhenChangingToAndFromAllPlayersMode()
-        {
-            Database.Setup(d => d.GetBeatmapChecksumAsync(3333)).ReturnsAsync("3333");
-            Database.Setup(d => d.GetBeatmapChecksumAsync(4444)).ReturnsAsync("4444");
-
-            // The room is free-for-all initially.
-            await Hub.JoinRoom(ROOM_ID);
-            await Hub.ChangeSettings(new MultiplayerRoomSettings { QueueMode = QueueMode.AllPlayers });
+            // User 1 adds an extra item
             await Hub.AddPlaylistItem(new MultiplayerPlaylistItem
             {
                 BeatmapID = 3333,
                 BeatmapChecksum = "3333"
             });
 
-            SetUserContext(ContextUser2);
-
-            await Hub.JoinRoom(ROOM_ID);
-            await Hub.AddPlaylistItem(new MultiplayerPlaylistItem
-            {
-                BeatmapID = 4444,
-                BeatmapChecksum = "4444"
-            });
+            checkCurrentItem(1);
+            checkOrder(1, 0);
+            checkOrder(2, 1);
 
+            // Item played.
             await runGameplay();
             checkCurrentItem(2);
+            checkOrder(2, 0);
 
-            await Hub.ChangeSettings(new MultiplayerRoomSettings { QueueMode = QueueMode.AllPlayersRoundRobin });
-            checkCurrentItem(3);
-
-            await Hub.ChangeSettings(new MultiplayerRoomSettings { QueueMode = QueueMode.AllPlayers });
-            checkCurrentItem(2);
-        }
-
-        [Fact]
-        public async Task CurrentItemOrderedByPlayedAt()
```

Comment:
I've removed these other tests because they specifically tested for jumping in the front of the queue. This class only contains one test now which is the full round robin sequence.

This test in particular is upheld by the last few lines at the bottom of the new test, that ensure the current item doesn't change after the lest item in the room is played.